### PR TITLE
Moving U-boot to v2022.10

### PR DESCRIPTION
**Pull request by Insei**
- Since version 2022.07, u-boot has added device tree sources for rpi4, and now we can use them, instead of the precompiled dtb.
- Adding support for overlays. U-boot builds dtb with a flag that adds a `__symbols__` node which is used in many overlays for rpi4.
- Update u-boot to the latest stable version 2022.10.
- Adapt old patches to the new version of u-boot.
- Remove the overlay with the addition of the smbios node because such a node [already exists](https://github.com/u-boot/u-boot/blob/master/arch/arm/dts/bcm283x-u-boot.dtsi#L10) in dtb which is built from the u-boot sources.
- Remove old patches for u-boot version v2020.07 and 2021.04

Additional information sent @rouming: The customer says that our EVE-OS does not load on his rpi4 with mmc errors.
The problem turned out to be that RPi introduced a new rev of the SoC in the RPi4 series of devices and adjusted some pieces of how bits of the SoC is wired up. The RPi400, and rev 1.4 and later of the RPi4b have this new rev of the SoC. The issue has been resolved in u-boot version 2022.10.
https://bugzilla.redhat.com/show_bug.cgi?id=2011423

### CODE REVIEW

**Comment by rouming** [comment]
@Insei @rene Thanks for separating uboot from tpm changes (#2896). Did you run an image with this pr on a rpi4? If yes and is all good, @eriknordmark can we merge this? If not yet tested - we can pull and run an image.

**Comment by rouming** [comment]
Great, thanks

File: pkg/u-boot/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN eve-alpine-deploy.sh
 
 SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
 
-ENV VERSION v2021.10
```
Comment (giggsoff) [inline pkg/u-boot/Dockerfile:8]: Should we remove patches for `v2021.10` as well? I think we can find them in git tree if needed.

**Review by giggsoff — COMMENTED** [review]
Should we keep `rpi/overlays/raspberrypi-uno-220.dts` and patches for old u-boot?
Please, adjust README.md as we have no patches for USB and no blobs in repo.

_Also I cannot understand, what we try to do in `0007-tweak-uefi-dt-selection.patch`, but seems it was here before, just renaming_.

**Comment by Insei** [comment]
> _Also I cannot understand, what we try to do in `0007-tweak-uefi-dt-selection.patch`, but seems it was here before, just renaming_.

We don't do anything with this patch, it just migrated to the new patch folder to the new version, and it was removed from the old patch folders. Such changes show up strangely in the PR changes, see the commits separately.

`rpi/overlays/raspberrypi-uno-220.dts` Needed for the Advantech UNO-220 platform, it overwrites the SMBIOS nodes for this platform. But as I understand it, this must be done manually in config.txt after writing the image to the sdcard. I think this will need to be specified in the Readme.

> Please, adjust README.md as we have no patches for USB and no blobs in repo
Nice catch. Will do it.

I also prefer to leave one revision of the old patches, for greater clarity, that when updating, no patches were lost, it's easier than looking at git history.

File: pkg/u-boot/Dockerfile (hunk 1 above)
Comment (Insei) [reply pkg/u-boot/Dockerfile:8]: @giggsoff I also prefer to leave one revision of the old patches, for greater clarity, that when updating, no patches were lost, it's easier than looking at git history.

**Comment by Insei** [comment]
Tested, everything is ok.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**